### DISCUSSION TITLE
why not custom channel name

### DISCUSSION THREAD

**Discussion by dreamfly32** [django/channels#1656] [Help!]
i try custom a channel name ,eg: "user01", when i channel_layer.send("user01", {
                    "type": "system.message",
                    "message": data
                })
nothing error,but "user01" did't recevie message
when i use yours self.channel_name,it is ok. why ?

**Comment by mondwan** [comment]
* You can take a look source codes in `channels.consumer.AsyncConsumer.__call__`
    * channel_name is generated on `channels.consumer.AsyncConsumer.__call__`
    * channel_receive() is an alias of `self.channel_layer.receive(self.channel_name)`
* So, i guess you need to call self.channel_layer.receive('user01') at all such that it can listen to the correct place
